src/amp-story-player/amp-story-player-impl.js

```diff
@@ -177,6 +178,54 @@ export class AmpStoryPlayer {
     this.layoutCallback();
   }
 
+  /**
+   * Adds stories to the player, both adding new anchor tags to the player,
+   * and creating or assigning iframes to those that are close to the
+   * current playing story.
+   *
+   * @param {!Array<!{href: string, title: ?string}>} stories
+   */
+  addStories(stories) {
+    for (let i = 0; i < stories.length; i++) {
+      const story = stories[i];
+      const anchor = this.createStoryAnchor_(story);
+
+      this.element_.append(anchor);
+      this.stories_.push(anchor);
```

Comment:
Maybe these are questions for Enrique but:
1. Why is the `this.stories_` array containing HTML elements rather than `{href, title}` objects?
2. Why do we need to create a HTML element from the configuration and append it in the DOM?

If `this.stories_` was an array of objects, we could just sanitize the data passed to `add(stories)` and push it into the array.

Reply:
I will talk to @Enriqe and follow up on this one.

Reply:
Done! Appending to the DOM has been removed on the latest commit

Reply:
I feel pretty strongly about the fact that `this.stories_` should be an array of objects with a clear typedef, so we know what it contains.
It has two benefits for this very PR:
- Now that you wrote validation for the `add()` method, the same validation can be used when retrieving the initial anchors (cool improvement)
- You don't have to create an anchor from the `add()` method

Reply:
Something to point out is that the anchors on `this.stories_` have things like the current iframe assigned to the story, which is something that is only handled internally. Hence, we would need to have a typedef for what we expect from the user to add the story, and a typedef for the story object internally.

Reply:
Good point. The validation is taken care of with your `isStoryObject` method. The typedef can have extra parameters used by the internal implementations, like:

```
/**
 * @typedef {{
 *   href: string,
 *   title: string,
 *   poster: ?string,
 *   iframeIdx: ?number
 * }}
 */
```

You'll have to initialize the `this.stories_` items with the extra `iframeIdx` parameter, but it's good practice to use objects that share the same prototype because browser engines can do further optimizations. (not that it matters for 10 items but 🤷‍♂️ )

Reply:
Makes sense! Let's have Enrique's input on this, but if we do end up making these changes, I think having a follow-up PR would be best, as there would be changes all around the code.

Reply:
Doing it as a followup sounds good to me, can you open a ticket?

Reply:
Absolutely, I got it :)

Reply:
@proyectoramirez can you share the link to the issue you filed?

Reply:
https://github.com/ampproject/amphtml/issues/29368#issue-661933898